src/main/scala/com/sky/kafka/configurator/KafkaTopicAdmin.scala

```diff
@@ -0,0 +1,71 @@
+package com.sky.kafka.configurator
+
+import java.util.concurrent.ExecutionException
+
+import cats.data.Reader
+import com.sky.kafka.configurator.error.TopicNotFound
+import org.apache.kafka.clients.admin.AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG
+import org.apache.kafka.clients.admin._
+import org.apache.kafka.common.config.ConfigResource
+import org.apache.kafka.common.errors.UnknownTopicOrPartitionException
+import org.zalando.grafter.{ Stop, StopResult }
+
+import scala.collection.JavaConverters._
+import scala.language.postfixOps
+import scala.util.{ Failure, Success, Try }
+
+object KafkaTopicAdmin {
+  def apply(adminClient: AdminClient): KafkaTopicAdmin = new KafkaTopicAdmin(adminClient)
+
+  def reader: Reader[AppConfig, KafkaTopicAdmin] = Reader { config =>
+    import com.sky.kafka.utils.MapToJavaPropertiesConversion.mapToProperties
+    KafkaTopicAdmin(AdminClient.create(Map(BOOTSTRAP_SERVERS_CONFIG -> config.bootstrapServers)))
+  }
+}
+
+class KafkaTopicAdmin(ac: AdminClient) extends TopicReader with TopicWriter with Stop {
+  override def fetch(topicName: String) = {
+
+    def topicDescription = Try(ac.describeTopics(Seq(topicName).asJava).values().get(topicName).get()) match {
+      case Success(result) => Success(result)
+      case Failure(e: ExecutionException) if e.getCause.isInstanceOf[UnknownTopicOrPartitionException] => Failure(TopicNotFound(topicName))
+      case other => other
+    }
+
+    def configDescription = Try{
+      ac.describeConfigs(Seq(configResourceForTopic(topicName)).asJava).all().get.get(configResourceForTopic(topicName)).entries().asScala.map { entry =>
```

Comment:
struggling to follow this line a bit. Assuming some of the calls made on this line can throw individually - would it make sense to split this into steps in the for comp below? Potentially the same for line 29 too

Reply:
There is only one method call throwing... the `.get` on the `KafkaFuture`, anyway I take that it is not straight forward to follow and will try to refactor.